concent_api/core/message_handlers.py

```diff
@@ -419,22 +411,52 @@ def handle_send_force_get_task_result(client_message: message.concents.ForceGetT
             reason=message.concents.ForceGetTaskResultRejected.REASON.AcceptanceTimeLimitExceeded,
         )
 
-    subtask = store_or_update_subtask(
-        task_id=task_to_compute.compute_task_def['task_id'],
+    subtask = get_one_or_none(
+        Subtask,
         subtask_id=task_to_compute.compute_task_def['subtask_id'],
-        provider_public_key=provider_public_key,
-        requestor_public_key=requestor_public_key,
-        state=Subtask.SubtaskState.FORCING_RESULT_TRANSFER,
-        next_deadline=(
-            int(task_to_compute.compute_task_def['deadline']) +
-            2 * maximum_download_time +
-            3 * settings.CONCENT_MESSAGING_TIME
-        ),
-        set_next_deadline=True,
-        report_computed_task=client_message.report_computed_task,
-        task_to_compute=task_to_compute,
-        force_get_task_result=client_message,
     )
+    if subtask is None:
+        subtask = store_subtask(
+            task_id=task_to_compute.compute_task_def['task_id'],
+            subtask_id=task_to_compute.compute_task_def['subtask_id'],
+            provider_public_key=provider_public_key,
+            requestor_public_key=requestor_public_key,
+            state=Subtask.SubtaskState.FORCING_RESULT_TRANSFER,
+            next_deadline=(
+                int(task_to_compute.compute_task_def['deadline']) +
+                2 * maximum_download_time +
+                3 * settings.CONCENT_MESSAGING_TIME
+            ),
+            task_to_compute=task_to_compute,
+            report_computed_task=client_message.report_computed_task,
+            force_get_task_result=client_message,
+        )
+
+    else:
+        if subtask.state_enum == Subtask.SubtaskState.FORCING_RESULT_TRANSFER:
+            return message.concents.ServiceRefused(
+                reason=message.concents.ServiceRefused.REASON.DuplicateRequest,
+            )
+        if subtask is not None:
+            if task_to_compute is not None and subtask.task_to_compute is not None:
```

Comment:
Can `subtask.task_to_compute` even be `None`? And can we even get this far in the handler without failing one of the validations earlier?

I think this whole condition should be an `assert` because it will never fail unless there's a serious bug somewhere. After you assert you can just call `validate_all_messages_identical()` without checking the condition.

Reply:
it can't be none, it was the whole block from other place, I moved it here, but forgot to remove unnecessary any more condition. Case ` if subtask is None:` is checked above and there is block for this situation. Here in else (subtask not none) assert is not necessary.